src/Random.hpp

```diff
@@ -1,11 +1,11 @@
 #pragma once
-#include <vector>
+#include <array>
 
-int random_range(int begin, int end);
+[[nodiscard]] int random_range(int begin, int end);
 
 template <typename T, typename... Args>
 T one_of(T first, Args... rest) {
-  T array[sizeof...(rest) + 1] = {first, rest...};
+  std::array<T, sizeof...(rest) + 1> array = {first, rest...};
 
-  return array[random_range(0, sizeof...(rest))];
+  return array.at(random_range(0, sizeof...(rest)));
```

Comment:
You know that this is huge performance degradation? Please use brackets, we don't need bounds check. Size is known at compile time.

Reply:
I thought that this is a minor performance degradation. I've run some benchmarks ([clang 12.0](https://quick-bench.com/q/HLDMU1dxB2ekimR1RLwsxJPeEDQ), [gcc 10.3](https://quick-bench.com/q/LJ0gD-NJLRJHkaomrg_xFFNmkh4)) and they do not show huge performance loss (however I assume it is not as straightforward as running this simple check). To be honest, I am not sure what should be the right approach, in cpp core guidelines we can find information about [using at](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#slcon3-avoid-bounds-errors). I've read that `gsl::at` checks are based on contracts that are implemented as macros, which can be disabled with the right switch (`GSL_UNENFORCED_ON_CONTRACT_VIOLATION `), however it turns off all contract checks, not only the ones related to bounds checking. It seems very strange to me that bound checks are not implemented as assertions which can be disabled in Release mode...

Reply:
Let's not overengineer here. 

We know the size at compile time, we don't need extra bounds check. 

Code readability is the same with at/[].

I don's see any cons.

In your benchmark you are not comparing access only, but also the construction of the array. Here is benchmark that check only the access: https://quick-bench.com/q/kt32INZXP_lwv9RLqT5lc31Sloo

Reply:
I've read also the discussion here: https://github.com/isocpp/CppCoreGuidelines/issues/1070. In general, I think we can preffer .at over the brackets, however here we know the bounds at compile time so we can keep [].

Reply:
Agreed!